### PR TITLE
Fix for `gr.File()` "stuck" uploading

### PR DESCRIPTION
**Pull request by dawoodkhan82**
## Description
Checks for file blob, and updates download url for file.

Please include a concise summary, in clear English, of the changes in this pull request. If it closes an issue, please mention it here.

Closes: #(issue)

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

**Comment by abidlabs** [comment]
@dawoodkhan82 very nice, this does fix the issue for me! However, I'm pretty sure that the data URLs have changed. When you used to click on them, you'd get a path to the temporary file on the server, now you get a more mysterious URL like: `blob:http://127.0.0.1:7864/6caa7c02-484d-4a0f-b2db-5fc6bd462df4`

**Comment by abidlabs** [comment]
Also is a complete demo working for you? here's what I tried (nothing seems to happen):

```py
import gradio as gr

with gr.Blocks() as demo:
    f = gr.File()
    f2 = gr.File()
    f.upload(lambda x:x.name, f, f2)
    
demo.launch()
```

**Comment by dawoodkhan82** [comment]
Oh I see, will take a look

File: js/file/shared/FilePreview.svelte [hunk 1]
```diff
@@ -33,9 +33,9 @@
 					</td>
 
 					<td class="download">
-						{#if file.data}
+						{#if file.data || file.blob}
```
Comment (freddyaboulton) [inline js/file/shared/FilePreview.svelte:36]: Why do we have blobs now? All files are uploaded/stored on the server. file.data should not be null 🤔

**Pull request closed without merging**